src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,50 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] result = new int[matrix.length];
+
+        for (int first = 0, second = 0; first < matrix.length; first++, second++) {
+            result[first] += matrix[first][second];
+        }
+
+        return result;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] result = new int[matrix.length];
+
+        for (int first = 0, second = matrix.length - 1; first < matrix.length; first++, second--) {
+            result[first] = matrix[first][second];
+        }
+
+        return result;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int minElement = matrix[0][0];
+
+        for (int row = 0; row < matrix.length; row++) {
+            for (int column = 0; column < matrix.length; column++) {
+                if (matrix[row][column] < minElement) {
+                    minElement = matrix[row][column];
+                }
+            }
+        }
+
+        return minElement;
     }
 
     public int getMax(int[][] matrix) {
-        return 0;
+        int maxElement = matrix[0][0];
+
+        for (int row = 0; row < matrix.length; row++) {
```

Comment:
FYI: 
in general it's ok to "i" and "j" naming if that's just about loops, that's common practice